transport-native-io_uring/src/test/java/io/netty/channel/uring/IoUringFileRegionTest.java

```diff
@@ -43,18 +47,34 @@ public void testSendFile() throws IOException, InterruptedException {
         File inFile = File.createTempFile(UUID.randomUUID().toString(), ".tmp");
         inFile.deleteOnExit();
         Files.write(inFile.toPath(), sampleString.getBytes());
-        SynchronousQueue<byte[]> sendFileResult = new SynchronousQueue<>();
+        SynchronousQueue<ByteBuf> sendFileResult = new SynchronousQueue<>();
 
         ServerBootstrap serverBootstrap = new ServerBootstrap();
         serverBootstrap.channel(IoUringServerSocketChannel.class);
         Channel serverChannel = serverBootstrap.group(group)
                 .childHandler(new ChannelInboundHandlerAdapter() {
+                    private CompositeByteBuf compositeByteBuf;
+
+                    @Override
+                    public void handlerAdded(ChannelHandlerContext ctx) throws Exception {
+                        compositeByteBuf = ctx.alloc().compositeBuffer();
+                    }
+
+                    @Override
+                    public void handlerRemoved(ChannelHandlerContext ctx) throws Exception {
+                        if (compositeByteBuf != null) {
+                            compositeByteBuf.release();
+                        }
+                    }
+
                     @Override
                     public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception {
                         ByteBuf buf = (ByteBuf) msg;
-                        byte[] bytes = new byte[buf.readableBytes()];
-                        buf.readBytes(bytes);
-                        sendFileResult.put(bytes);
+                        compositeByteBuf.addComponent(true, buf);
+                        if (compositeByteBuf.readableBytes() == inFile.length()) {
+                            sendFileResult.put(compositeByteBuf);
```

Comment:
Maybe the sending should be in `channelReadComplete`?

Reply:
why ?

Reply:
Hmm… right, no need with the file length check.